From 51e3c8302838a89b854085195725d1c6d102aff4 Mon Sep 17 00:00:00 2001
From: Kiran Kumar K <kirankumark@marvell.com>
Date: Thu, 5 Sep 2019 11:20:18 +0530
Subject: [PATCH 0284/1921] octeontx2-af: match nvgre as ltype

Add change to match NVGRE as ltype for RSS in sync with KPU profile.

Change-Id: I7ee72266dab30d60bb9d9a281ee49f73ee5d2657
Signed-off-by: Kiran Kumar K <kirankumark@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/15626
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
index a6a0b754055e..b99f9571697c 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
@@ -2648,7 +2648,7 @@ static int set_flowkey_fields(struct nix_rx_flowkey_alg *alg, u32 flow_cfg)
 			field->lid = NPC_LID_LD;
 			field->hdr_offset = 4; /* VSID offset */
 			field->bytesm1 = 2;
-			field->ltype_match = NPC_LT_LD_GRE;
+			field->ltype_match = NPC_LT_LD_NVGRE;
 			field->ltype_mask = 0xF;
 			break;
 		case NIX_FLOW_KEY_TYPE_VXLAN:
-- 
2.31.1

